### PR TITLE
Create HeaderButton QML component

### PR DESCRIPTION
**Pull request by nicolasfella**
Deduplicate the Button code by moving it into a separate file.

Also use the same background code for the apps button since as far as I can tell it didn't behave any different

Signed-off-by: Nicolas Fella <nicolas.fella@gmx.de>

### CODE REVIEW

**Comment by DominiqueFuchs** [comment]
Hi @nicolasfella thanks for your proposals on this! I generally agree with you, but would like to wait until

- Some of the pending PRs are merged, which partially already adress this
- The codebase of this is a little bit less changing, I just made several design changes due to ongiong development and discussion on this

I'll get back to this very soon and will happily integrate this!

File: src/gui/tray/HeaderButton.qml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    display: AbstractButton.IconOnly
+    Layout.preferredWidth:  Style.trayWindowHeaderHeight
+    Layout.preferredHeight: Style.trayWindowHeaderHeight
+    flat: true
+
+    icon.width: Style.headerButtonIconSize
+    icon.height: Style.headerButtonIconSize
+    icon.color: "transparent"
+
+    MouseArea {
```
Comment (er-vin) [inline src/gui/tray/HeaderButton.qml:25]: BTW, I know you're moving code around but this MouseArea can likely go away, Button has the click and hovering feature.

**Comment by er-vin** [comment]
> Hi @nicolasfella thanks for your proposals on this! I generally agree with you, but would like to wait until
> 
>     * Some of the pending PRs are merged, which partially already adress this
> 
>     * The codebase of this is a little bit less changing, I just made several design changes due to ongiong development and discussion on this
> 
> 
> I'll get back to this very soon and will happily integrate this!

That was a little while ago now. What is blocking this PR and the other ones from @nicolasfella exactly? Asking because it's pretty much some of the code changes I had in mind myself when looking at the code last week. :-)

**Comment by DominiqueFuchs** [comment]
@er-vin I‘m not aware to ever have used the term of „blocking“ here - must have been unintentional then. My previous comment was rather a suggestion based on the assumption that the other PRs will be merged soon as we already talkes about another beta. At least #1940 and #1939 are still pending I think (mobile rn).

tl;dr: Feel free to merge and modify whatever you want.

**Comment by er-vin** [comment]
Alright, so let's fast track #1940 and #1939 just went ahead and reviewed. As soon as they're in we'll need to properly rebase @nicolasfella work.

**Comment by sbeyer** [comment]
`Window.qml` and `UserLine.qml` are for some reason listed in `${client_UI_SRCS}` defined in `src/gui/CMakeLists.txt`. Could it be that your new QML component's file should be mentioned there, too? Interestingly, I could remove the whole `client_UI_SRCS` definition and it still compiled and worked. I have no idea why.

**Comment by sbeyer** [comment]
> As soon as they're in we'll need to properly rebase @nicolasfella work.

FYI: In my forked repository I maintain "my" master branch where I merge some not-yet-merged branches or WIP branches to see how it works out all together. Currently it contains #1939, #2058, #2059 that all conflict with this branch. Also git presents the merge conflicts a little hazy due to changed indentation for `onClicked`. So, to save someone (@nicolasfella) from the work of resolving these conflicts again, [here is the resulting file](https://github.com/sbeyer/nextcloud-desktop/blob/a6e2d958c72c629c5526a0efb797ec058c061550/src/gui/tray/Window.qml?raw=true) and [here is the diff](https://github.com/sbeyer/nextcloud-desktop/commit/a6e2d958c72c629c5526a0efb797ec058c061550#diff-7986023a889dfd81450ec5fc086c7e94).

**Comment by er-vin** [comment]
> `Window.qml` and `UserLine.qml` are for some reason listed in `${client_UI_SRCS}` defined in `src/gui/CMakeLists.txt`. Could it be that your new QML component's file should be mentioned there, too? Interestingly, I could remove the whole `client_UI_SRCS` definition and it still compiled and worked. I have no idea why.

This looks like a left over from before CMAKE_AUTOUIC existed. Nowadays CMake automatically harvest for .ui files if it sees corresponding includes in the code. Putting qml files in such a list looks like a mistake.
